### PR TITLE
Upgrade to Xtext 2.27.0 and Lsp4j 0.14.0

### LINKED ISSUE #1233: Error when running ./gradlew runLfc (Could not determine the dependencies)
I encountered a weird error when running `./gradlew runLfc` recently.
(I remember running it fine last week)

Below is the raw output from the `./gradlew runLfc` command:

```
$ ./gradlew runLfc --args test/C/src/After.lf

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':org.lflang:generateXtext'.
> Could not resolve all files for configuration ':org.lflang:compileClasspath'.
   > Could not find any matches for org.osgi.service:org.osgi.service.prefs:[1.1.0,1.2.0) as no versions of org.osgi.service:org.osgi.service.prefs are available.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/osgi/service/org.osgi.service.prefs/maven-metadata.xml
       - file:/Users/hokeunkim/.m2/repository/org/osgi/service/org.osgi.service.prefs/
     Required by:
         project :org.lflang > org.eclipse.platform:org.eclipse.core.runtime:3.24.0 > org.eclipse.platform:org.eclipse.equinox.preferences:3.10.0

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1
```

Would anyone have any ideas about what's going on here?

My computer is M1 Macbook Pro with OS X Monterey (Version 12.3.1). It ran fine last week.
Java version: 
```
$java --version

openjdk 17.0.3 2022-04-19
OpenJDK Runtime Environment Homebrew (build 17.0.3+0)
OpenJDK 64-Bit Server VM Homebrew (build 17.0.3+0, mixed mode, sharing)
```

I would really appreciate your help.

### PR DESCRIPTION
**Pull request by lhstrh**
Upgrade xtext from `2.26.0` to `2.27.0`, lsp4j to `0.14.0`, and eclipse runtime to `3.25.0`.

Some other version changed as well, which is the result of running `bin/bump-versions --all`, which apparently hadn't been done the last time version numbers were changed in `gradle.properties`.

### CODE REVIEW

File: org.lflang.tests/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -7,13 +7,13 @@ Bundle-Version: 0.2.2.qualifier
...
+ org.junit.jupiter.api;bundle-version="5.8.2",
  org.eclipse.xtext.testing,
  org.eclipse.xtext.xbase.testing,
- org.eclipse.xtext.xbase.lib;bundle-version="2.26.0",
+ org.eclipse.xtext.xbase.lib;bundle-version="2.27.0",
  javax.annotation;bundle-version="1.2.0",
  org.eclipse.core.jobs,
  org.opentest4j;bundle-version="1.2.0",
- org.eclipse.lsp4j;bundle-version="0.10.0"
+ org.eclipse.lsp4j;bundle-version="0.12.0"
```
Comment (petervdonovan) [inline org.lflang.tests/META-INF/MANIFEST.MF:17]: It's a little weird that the LSP4J version was lagging behind in the tests package. It makes me wonder if LSP4J is even needed in there.

**Review by petervdonovan — APPROVED** [review]
I see nothing wrong with this as long as the tests pass.

File: org.lflang.tests/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -7,13 +7,13 @@ Bundle-Version: 0.2.2.qualifier
 Bundle-SymbolicName: org.lflang.tests; singleton:=true
 Bundle-ActivationPolicy: lazy
 Require-Bundle: org.lflang,
- org.junit.jupiter.api;bundle-version="[5.0.0,6.0.0)",
+ org.junit.jupiter.api;bundle-version="5.8.2",
```
Comment (lhstrh) [inline org.lflang.tests/META-INF/MANIFEST.MF:10]:
```suggestion
 org.junit.jupiter.api;bundle-version="[5.0.0,6.0.0)",
```

File: org.lflang.ui.tests/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -7,7 +7,7 @@ Bundle-Version: 0.2.2.qualifier
 Bundle-SymbolicName: org.lflang.ui.tests; singleton:=true
 Bundle-ActivationPolicy: lazy
 Require-Bundle: org.lflang.ui,
- org.junit.jupiter.api;bundle-version="[5.0.0,6.0.0)",
+ org.junit.jupiter.api;bundle-version="5.8.2",
```
Comment (lhstrh) [inline org.lflang.ui.tests/META-INF/MANIFEST.MF:10]:
```suggestion
 org.junit.jupiter.api;bundle-version="[5.0.0,6.0.0)",
```

File: org.lflang.tests/META-INF/MANIFEST.MF (hunk 1 above)
Comment (lhstrh) [reply org.lflang.tests/META-INF/MANIFEST.MF:17]: Yeah, it's a little weird.

**Comment by lhstrh** [comment]
It's unclear to me why Maven/Tycho cannot find `org.eclipse.xtext.xbase.lib`. It appears to be published here:

https://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.27.0/features/

**Comment by lhstrh** [comment]
> It's unclear to me why Maven/Tycho cannot find `org.eclipse.xtext.xbase.lib`. It appears to be published here:
> 
> https://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.27.0/features/

@a-sr: help? :fear:

**Comment by a-sr** [comment]
Sorry for my late reply. I must have missed the GH notification.
The problem is that increasing version numbers in the pom is not sufficient for the maven tycho build. You need to adjust the update side in the target platform definition as well, if you switch to a newer Eclipse or Xtext version (same as in #1021). Otherwise the dependencies cannot be resolved in the Epoch build.
I pushed a fix for that and also updated the oomph setup.

**Comment by a-sr** [comment]
I had to downgrade the junit.jupiter dependency to 5.8.1 because 5.8.2 is not available via the eclipse update site. The version of the mwe launcher also needed adjustment because the latest Eclipse release only provides a newer version.

I added a few comments to the pom file that point out that the update sites in the target platform need to adjusted when changing versions.

**Comment by lhstrh** [comment]
Thanks for these changes (and the clarifying comments), @a-sr!

File: org.lflang.diagram/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -12,13 +12,11 @@ Require-Bundle: de.cau.cs.kieler.klighd;bundle-version="2.0.0",
  com.google.guava,
  org.eclipse.xtext,
  org.eclipse.xtext.xbase.lib,
- org.eclipse.elk.core;bundle-version="0.8.1",
- org.eclipse.elk.alg.layered;bundle-version="0.8.1",
- org.eclipse.core.resources;bundle-version="3.13.900",
+ org.eclipse.core.resources;bundle-version="3.16.0",
```
Comment (lhstrh) [inline org.lflang.diagram/META-INF/MANIFEST.MF:15]:
```suggestion
 org.eclipse.elk.core;bundle-version="0.8.1",
 org.eclipse.elk.alg.layered;bundle-version="0.8.1",
 org.eclipse.core.resources;bundle-version="3.16.0",
```

**Pull request merged**